models/db/setting.go

```diff
@@ -0,0 +1,222 @@
+// Copyright 2023 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package db
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"code.gitea.io/gitea/modules/cache"
+	setting_module "code.gitea.io/gitea/modules/setting"
+	"code.gitea.io/gitea/modules/timeutil"
+	"code.gitea.io/gitea/modules/util"
+
+	"xorm.io/builder"
+)
+
+type ResourceSetting struct {
+	ID           int64              `xorm:"pk autoincr"`
+	GroupID      int64              `xorm:"index unique(key_repoid)"`               // to load all of some group's settings
+	SettingKey   string             `xorm:"varchar(255) index unique(key_groupid)"` // ensure key is always lowercase
+	SettingValue string             `xorm:"text"`
+	Version      int                `xorm:"version"` // prevent to override
+	Created      timeutil.TimeStamp `xorm:"created"`
+	Updated      timeutil.TimeStamp `xorm:"updated"`
+}
+
+func (s *ResourceSetting) AsBool() bool {
+	if s == nil {
+		return false
+	}
+
+	b, _ := strconv.ParseBool(s.SettingValue)
+	return b
+}
+
+// GetSettings returns specific settings
+func GetSettings(ctx context.Context, tableName string, groupID int64, keys []string) (map[string]*ResourceSetting, error) {
+	for i := 0; i < len(keys); i++ {
```

Comment:
I think that's one of the first times I've seen a "normal" C for-loop here.
```suggestion
	for i := range keys; {
```